pkg/handlers/s3_dev_handler.go

```diff
@@ -165,22 +165,20 @@ func (h *s3DevHandler) OnAdd(ns string, vc kvcv1.VolumeConfig, controllerRef met
 }
 
 func (h *s3DevHandler) OnDelete(ns string, vc kvcv1.VolumeConfig, controllerRef metav1.OwnerReference) {
-	for _, client := range h.k8sResourceClients {
-		if client.Plural() == "nodes" {
-			continue
-		}
+	podClient := getK8SResourceClientFromPlural(h.k8sResourceClients, "pods")
```

Comment:
We also need to delete the actual data on the node. That can go in a follow-up though.

Reply:
Agreed. But let's add it in a follow-up PR with the logic for garbage collection.